src/sentry/migrations/0349_actor_user_direct_reference.py

```diff
@@ -0,0 +1,50 @@
+# Generated by Django 2.2.28 on 2023-01-06 18:55
+
+import django.db.models.deletion
+from django.conf import settings
+from django.db import migrations
+
+import sentry.db.models.fields.foreignkey
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production. For
+    # the most part, this should only be used for operations where it's safe to run the migration
+    # after your code has deployed. So this should not be used for most operations that alter the
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = False
+
+    dependencies = [
+        ("sentry", "0348_add_outbox_and_tombstone_tables"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="externalactor",
+            name="team",
+            field=sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                null=True, on_delete=django.db.models.deletion.CASCADE, to="sentry.Team"
+            ),
+        ),
+        migrations.AddField(
+            model_name="externalactor",
+            name="user",
+            field=sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                null=True, on_delete=django.db.models.deletion.CASCADE, to=settings.AUTH_USER_MODEL
+            ),
+        ),
+        migrations.AddField(
+            model_name="grouphistory",
+            name="user",
+            field=sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                null=True, on_delete=django.db.models.deletion.SET_NULL, to=settings.AUTH_USER_MODEL
+            ),
```

Comment:
Should we be adding another foreign key here? This looks like a key that would need to be broken in the future as it is between user (control silo) and group history (region silo).

Reply:
See comment above.